packages/pages/src/dev/client/entry-client.tsx

```diff
@@ -9,7 +9,7 @@ const hydrate = async () => {
   };
 
   // Can't use string interpolation here so src/templates is hardcoded
-  const templates = import.meta.glob("/src/templates/*.(jsx|tsx)");
+  const templates = import.meta.glob("/src/templates/**/*.(jsx|tsx)");
 
   const routes: Route[] = Object.keys(templates).map((path) => {
```

Comment:
this routes variable is a little weird, it looks like we only need the single template variable we get in the `routes.find` below? but maybe we can address this in the cleanup epic

Reply:
yea I took a close look at the logic there, most of the code here is not really needed. `import.meta.glob("/src/templates/*.(jsx|tsx)");` would give use a `templates` object like this ([Vite doc](https://vitejs.dev/guide/features.html#glob-import)):
```
// code produced by vite
const modules = {
  '/src/templates//foo.tsx': () => import('./dir/foo.js'),
  '/src/templates//bar.tsx': () => import('./dir/bar.js')
}
```
since we already have the template path, we can just directly import that.

Reply:
mm actually, I see that there's a limitation with vite and dynamic import:
<img width="1174" alt="Screen Shot 2022-10-21 at 1 54 51 PM" src="https://user-images.githubusercontent.com/36055303/197258889-3db3c21e-5b26-41c9-9288-e4c3fcb07fa0.png">
I will revert the changes, and perhaps we can address this in the cleanup epic

Reply:
a little out of scope it'd be nice to get that cleaned up. right now the code is assuming `src/templates` is the root of where templates will be which prevents us from allowing that to be configurable.

Reply:
for sure! As this item is blocking other ones in the epic, I think we can make a separate item to address this existing bug.

Reply:
https://yexttest.atlassian.net/browse/SLAP-2426